src/diem/diem_types/__init__.py

```diff
@@ -405,6 +412,100 @@ def bcs_deserialize(input: bytes) -> "RawTransaction":
         return v
 
 
+class RefundMetadata:
+    VARIANTS = []  # type: typing.Sequence[typing.Type[RefundMetadata]]
+
+    def bcs_serialize(self) -> bytes:
+        return bcs.serialize(self, RefundMetadata)
+
+    @staticmethod
+    def bcs_deserialize(input: bytes) -> "RefundMetadata":
+        v, buffer = bcs.deserialize(input, RefundMetadata)
+        if buffer:
+            raise st.DeserializationError("Some input bytes were not read")
+        return v
+
+
+@dataclass(frozen=True)
+class RefundMetadata__RefundMetadataV0(RefundMetadata):
+    INDEX = 0  # type: int
+    value: "RefundMetadataV0"
+
+
+RefundMetadata.VARIANTS = [
+    RefundMetadata__RefundMetadataV0,
+]
+
+
+@dataclass(frozen=True)
+class RefundMetadataV0:
+    transaction_version: st.uint64
+    reason: "RefundReason"
+
+    def bcs_serialize(self) -> bytes:
+        return bcs.serialize(self, RefundMetadataV0)
+
+    @staticmethod
+    def bcs_deserialize(input: bytes) -> "RefundMetadataV0":
+        v, buffer = bcs.deserialize(input, RefundMetadataV0)
+        if buffer:
+            raise st.DeserializationError("Some input bytes were not read")
+        return v
+
+
+class RefundReason:
+    VARIANTS = []  # type: typing.Sequence[typing.Type[RefundReason]]
+
+    def bcs_serialize(self) -> bytes:
+        return bcs.serialize(self, RefundReason)
+
+    @staticmethod
+    def bcs_deserialize(input: bytes) -> "RefundReason":
+        v, buffer = bcs.deserialize(input, RefundReason)
+        if buffer:
+            raise st.DeserializationError("Some input bytes were not read")
+        return v
+
+
+@dataclass(frozen=True)
+class RefundReason__OtherReason(RefundReason):
+    INDEX = 0  # type: int
+    pass
+
+
+@dataclass(frozen=True)
+class RefundReason__InvalidReferenceId(RefundReason):
+    INDEX = 1  # type: int
+    pass
+
+
+@dataclass(frozen=True)
+class RefundReason__DuplicateReferenceId(RefundReason):
+    INDEX = 2  # type: int
+    pass
+
+
+@dataclass(frozen=True)
+class RefundReason__UserInitiatedPartialRefund(RefundReason):
+    INDEX = 3  # type: int
+    pass
+
+
+@dataclass(frozen=True)
+class RefundReason__UserInitiatedFullRefund(RefundReason):
+    INDEX = 4  # type: int
+    pass
+
+
+RefundReason.VARIANTS = [
+    RefundReason__OtherReason,
+    RefundReason__InvalidReferenceId,
+    RefundReason__DuplicateReferenceId,
```

Comment:
Can we tweak this based upon this: https://github.com/diem/dip/pull/135

these don't really exist yet...

pr looks great otherwise